### PR TITLE
fix: wallet show coalesced updates

### PR DESCRIPTION
**Pull request by turadg**
refs: #6290 

## Description

Main change is to make the coalesceUpdates utility track the status of invitations, in support of #6290. While in there I noticed the updater was getting a list going back in time, so the "last wins" was actually "latest wins". No bueno. This reverses the updates list before coalescing.

Also refactored the CLI to use casting and cleaned up some logging and types.

### Security Considerations
--
### Documentation Considerations

--

### Testing Considerations

These are UIs that don't have automated tests. Tested manually.

### CODE REVIEW

File: packages/agoric-cli/src/lib/rpc.js [hunk 1]
```diff
@@ -236,20 +235,23 @@ harden(storageHelper);
...
       const parts = storageHelper.unserializeTxt(content, ctx).at(-1);
-
+      for (const [name, remote] of parts) {
+        reverse[remote.boardId] = name;
+      }
       return [kind, Object.fromEntries(parts)];
     }),
   );
-  return Object.fromEntries(entries);
+  return { ...Object.fromEntries(entries), reverse };
```
Comment (dckc) [inline packages/agoric-cli/src/lib/rpc.js:254]: this puts mappings in both directions in the same object? That seems surprising.

File: packages/smart-wallet/src/utils.js [hunk 2]
```diff
@@ -29,7 +28,7 @@ export const makeWalletStateCoalescer = () => {
         if (currentAmount.brand.boardId === 'board02810') {
```
Comment (dckc) [inline packages/smart-wallet/src/utils.js]: The zoe invitation brand has `board02810`? give it a constant name?

File: packages/zoe/src/zoeService/offer/burnInvitation.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ import { E } from '@endo/eventual-send';
 export const burnInvitation = (invitationIssuer, invitation) => {
   const handleRejected = reason => {
     const err = assert.error(
-      X`A Zoe invitation is required, not ${invitation}`,
+      X`A Zoe invitation is required, not ${q(invitation)}`,
```
Comment (dckc) [inline packages/zoe/src/zoeService/offer/burnInvitation.js]: This seems risky - we don't know what it is we're divulging here.

File: packages/casting/src/follower-cosmjs.js [hunk 4]
```diff
@@ -82,7 +82,7 @@ const proofs = ['strict', 'none', 'optimistic'];
  * @param {any} sourceP
  * @param {import('./types').LeaderOrMaker} [leaderOrMaker]
  * @param {import('./types').FollowerOptions} [options]
- * @returns {Follower<FollowerElement<T>>}
```
Comment (dckc) [inline packages/casting/src/follower-cosmjs.js:85]: this didn't turn out to be quite right...

returns {ValueFollower<T>} and FollowerElement is a ValueFollowerElement

**Review by dckc — APPROVED** [review]
please consider tweaks as discussed

File: packages/zoe/src/zoeService/offer/burnInvitation.js (hunk 3 above)
Comment (turadg) [reply packages/zoe/src/zoeService/offer/burnInvitation.js]: removed

File: packages/smart-wallet/src/utils.js (hunk 2 above)
Comment (turadg) [reply packages/smart-wallet/src/utils.js]: changed to get `allegedInvitationBrand` from the petname

**Pull request merged**